comp/logs/agent/mock.go

```diff
@@ -37,6 +39,18 @@ func newMock(deps dependencies) optional.Option[Mock] {
 	return optional.NewOption[Mock](logsAgent)
 }
 
+// NewMock can be used in other packages using the log agent as a dependency.
+func NewMock(logSources *sources.LogSources) optional.Option[Component] {
```

Comment:
instead of adding a new `NewMock` method (duplicating what the other new mock does), maybe we can re-use what we have in line with how the component framework works. 

I tried to type this as feedback, but it ended up being too big for a comment so I just prototyped it out myself real quick: https://github.com/DataDog/datadog-agent/commit/099a6c1bea62548d9b401ffd9364ab021c1becf0

Let me know what you think or if you have any questions. The idea is to add mock methods to our mock implementation that allows you to mutate it's state, and then use it as a valid component instance. 

(feel free to take my code as is - but it may need a little cleanup)